libs/RobotKit/include/RobotController.h

```diff
@@ -288,25 +288,51 @@ class RobotController : public interface::RobotController
 		raise(system::robot::sm::event::autonomous_activities_mode_requested {});
 	}
 
+	void raiseAutonomousActivityModeExited() { raise(system::robot::sm::event::autonomous_activities_mode_exited {}); }
+
 	void onMagicCardAvailable(const MagicCard &card)
 	{
+		using namespace std::chrono;
+
+		static auto start = rtos::Kernel::Clock::now();
+		static auto stop  = rtos::Kernel::Clock::now();
+
+		auto is_playing			= _activitykit.isPlaying();
+		auto is_autonomous_mode = state_machine.is(system::robot::sm::state::autonomous_activities);
+
 		if (card == MagicCard::emergency_stop) {
 			raiseEmergencyStop();
 			return;
 		}
 
 		if (card == MagicCard::dice_roll) {
-			raiseAutonomousActivityModeRequested();
-			if (_activitykit.isPlaying()) {
-				_activitykit.stop();
+			if (!is_autonomous_mode) {
+				start = rtos::Kernel::Clock::now();
+
+				if (start - stop > 1s) {
+					raiseAutonomousActivityModeRequested();
+				}
+			}
+
+			if (is_autonomous_mode && !is_playing) {
+				stop = rtos::Kernel::Clock::now();
+
+				if (stop - start > 2s) {
+					raiseAutonomousActivityModeExited();
+				}
+			}
+
+			if (is_autonomous_mode && is_playing) {
+				start = rtos::Kernel::Clock::now();
+
+				if (start - stop > 1s) {
+					raiseAutonomousActivityModeRequested();
+				}
 			}
```

Comment:
Does this still work if not using `else if`? Validation on robot will confirm this.